plutus-contract/src/Language/Plutus/Contract/Tx.hs

```diff
@@ -77,14 +78,16 @@ emptyTx = UnbalancedTx mempty mempty mempty mempty I.always
 --   does not have any signatures, and is potentially unbalanced (ie. invalid).
 --   To produce a balanced 'Tx', use 'Language.Plutus.Contract.Wallet.balanceTx'.
 toLedgerTx :: UnbalancedTx -> L.Tx
-toLedgerTx utx = L.Tx
+toLedgerTx utx =
+    let tx = L.Tx
             { L.txInputs = _inputs utx
             , L.txOutputs = _outputs utx
             , L.txForge = _forge utx
             , L.txFee = 0
             , L.txValidRange = _validityRange utx
             , L.txSignatures = Map.empty
             }
+     in tx { L.txFee = minFee tx }
```

Comment:
The fee should be added after the transaction has been balanced, so it's ok to leave it at 0 here (see also https://github.com/input-output-hk/plutus/issues/1320)

Reply:
If you add the transaction fee, then you will probably need to rebalance the transaction.

It is also a bit weird. If the user should use mostly (or only?) unbalanced transactions, then he won't be able to easily specify a custom fee. In order to do that he would need to balance the transaction, then add the fee, and rebalance the (already balanced?) transaction.

I think unbalanced should include any extra transaction fees that the user wants to pay, then during balancing the fees are allocated together with everything else.

Reply:
The fee depends on the size of the transaction (number of inputs and outputs, script size) so when balancing adds new inputs/outputs the fee might change. So I think it makes sense to compute the fee first, before balancing, but then the fee should be higher than the absolute minimum, to account for added inputs & outputs. Having a non-zero fee can be done in a later PR though.